test/StableBufferTest.mo

```diff
@@ -93,6 +93,27 @@ do {
   assert (B.toVarArray(c).size() == 2);
 };
 
+// test remove
+do {
+  let arr = [1,2,3,4,5,6,7,8,9,10];
+  let d = B.fromArray<Nat>(arr);
+  assert(B.remove(d, 1) == 2);
+  assert(B.remove(d, 4) == 6);
+  assert(B.remove(d, 0) == 1);
+  assert(B.remove(d, 6) == 10);
+  let expected = [3, 4, 5, 7, 8, 9];
+  assert (natIterEq(B.vals<Nat>(d), expected.vals()));
+  
+  // remove 4 more elements to trigger downsizing.
+  assert(B.remove(d, 1) == 4);
+  assert(B.remove(d, 0) == 3);
+  assert(B.remove(d, 2) == 8);
+  assert(B.remove(d, 1) == 7);
+
+  let expectedAfterDownsize = [5, 9];
+  assert (natIterEq(B.vals<Nat>(d), expectedAfterDownsize.vals()));
```

Comment:
Looks good! Just one tiny issue - this actually doesn't trigger downsizing yet.

`10 / 4 = 2`, and the condition is `if ((buffer.count - 1) : Nat < currentCapacity / DECREASE_THRESHOLD)`, so you'll need to remove one additional item to actually trigger the downsizing.

You can actually test what's being held in the buffer by doing an assert on `d.elems` (the internal var array structure).

So if you just add

```
assert(B.remove(d, 0) == 5);
assert (natIterEq(B.vals<Nat>(d), [9].vals()));
// test downsizing worked
assert (natIterEq(B.elems<Nat>(d), [9, 5, 5, 5, 5].vals());
```

This will actually test that the internal array size was halved.